### PR TITLE
Raises errors if attempt to use function or data backgrounds

### LINKED ISSUE #245: Data and function backgrounds/resolutions are not yet supported
Data and Function type backgrounds and resolutions are available in RAT, but not yet supported. We should raise an error when either of these are selected in both MATLAB and python.

### PR DESCRIPTION
**Pull request by alexhroom**
Fixes #245 by adding a new exception type `notImplemented` and raises it if the user attempts to use a data or function background.

Presently, the relevant test parts have been commented out. If someone thinks it's better to remove them entirely, please let me know.

### CODE REVIEW

File: tests/testBackgroundsClass.m [hunk 1]
```diff
@@ -82,19 +85,19 @@ function testAddBackground(testCase)
...
-                              'addBackground method not working');
-         testCase.verifyError(@() testCase.background.addBackground('New', 'fixed'), exceptions.invalidOption.errorID);
-         testCase.verifyError(@() testCase.background.addBackground('New', allowedTypes.Constant), exceptions.invalidNumberOfInputs.errorID);
-         testCase.verifyError(@() testCase.background.addBackground('New', allowedTypes.Constant.value, 6), exceptions.indexOutOfRange.errorID);
+         %testCase.background.addBackground(testCase.backgrounds{3, 1:5});
+         %testCase.verifyEqual(string(testCase.background.backgrounds.varTable{end, :}),...
+         %                     string(testCase.backgrounds(3, :)), 'addBackground method not working');
+         %testCase.background.addBackground(testCase.backgrounds{4, :});
+         %testCase.verifyEqual(string(testCase.background.backgrounds.varTable{end, :}),...
+         %                     string(testCase.backgrounds(4, :)), 'addBackground method not working');
```
Comment (DrPaulSharp) [inline tests/testBackgroundsClass.m:93]: Given the changes to the test input data, can these lines be uncommented?

File: tests/testBackgroundsClass.m [hunk 2]
```diff
@@ -82,19 +85,19 @@ function testAddBackground(testCase)
...
+         %testCase.background.addBackground('background 5', allowedTypes.Function, 'function_name', 1, 3);
+         %testCase.verifyEqual(string(testCase.background.backgrounds.varTable{end, :}),...
+         %                     ["background 5", string(allowedTypes.Function.value), "function_name", "background param 1", "background param 3", "", ""], ...
+         %                     'addBackground method not working');
+         %testCase.verifyError(@() testCase.background.addBackground('New', 'fixed'), exceptions.invalidOption.errorID);
+         %testCase.verifyError(@() testCase.background.addBackground('New', allowedTypes.Constant), exceptions.invalidNumberOfInputs.errorID);
+         %testCase.verifyError(@() testCase.background.addBackground('New', allowedTypes.Constant.value, 6), exceptions.indexOutOfRange.errorID);
```
Comment (DrPaulSharp) [inline tests/testBackgroundsClass.m:100]: The three lines here testing the errors can be uncommented too.

File: tests/testBackgroundsClass.m [hunk 3]
```diff
@@ -132,14 +135,14 @@ function testSetBackground(testCase)
 
          testCase.background.setBackground(1, 'type', allowedTypes.Constant.value);
          testCase.verifyEqual(testCase.background.backgrounds.varTable{1, 2}, string(allowedTypes.Constant.value), 'setBackground method not working');
-         testCase.background.setBackground('Background 1', 'type', allowedTypes.Function); 
-         testCase.verifyEqual(testCase.background.backgrounds.varTable{1, 2}, string(allowedTypes.Function.value), 'setBackground method not working');
-         testCase.verifyError(@() testCase.background.setBackground(2, 'type', 'random'), exceptions.invalidOption.errorID);
+         %testCase.background.setBackground('Background 1', 'type', allowedTypes.Function); 
+         %testCase.verifyEqual(testCase.background.backgrounds.varTable{1, 2}, string(allowedTypes.Function.value), 'setBackground method not working');
+         %testCase.verifyError(@() testCase.background.setBackground(2, 'type', 'random'), exceptions.invalidOption.errorID);
```
Comment (DrPaulSharp) [inline tests/testBackgroundsClass.m:140]: Uncomment this line.

File: tests/testBackgroundsClass.m [hunk 4]
```diff
@@ -132,14 +135,14 @@ function testSetBackground(testCase)
...
          
-         testCase.background.setBackground(3, 'Value1', 'random');
-         testCase.verifyEqual(testCase.background.backgrounds.varTable{3, 3}, "random", 'setBackground method not working');
-         testCase.background.setBackground('Background 3', 'Value1', 'Background param 1');
-         testCase.verifyEqual(testCase.background.backgrounds.varTable{3, 3}, "Background param 1", 'setBackground method not working');
+         %testCase.background.setBackground(3, 'Value1', 'random');
+         %testCase.verifyEqual(testCase.background.backgrounds.varTable{3, 3}, "random", 'setBackground method not working');
+         %testCase.background.setBackground('Background 3', 'Value1', 'Background param 1');
+         %testCase.verifyEqual(testCase.background.backgrounds.varTable{3, 3}, "Background param 1", 'setBackground method not working');
```
Comment (DrPaulSharp) [inline tests/testBackgroundsClass.m:145]: and these lines.

**Review by DrPaulSharp — COMMENTED** [review]
This looks good, please review `testBackgrounds.m` to see if some of the tests can be re-introduced.

File: tests/testBackgroundsClass.m (hunk 4 above)
Comment (alexhroom) [reply tests/testBackgroundsClass.m:145]: can't uncomment 142-143 as i'd changed background 3 to a constant background - Value1 of a constant background is validated but Value1 of a function background is not, so you can't just arbitrarily set Value1 for a test unless the background type is function

**Review by DrPaulSharp — APPROVED** [review]

**Pull request merged**